raiden/transfer/mediated_transfer/mediator.py

```diff
@@ -377,31 +377,32 @@ def set_expired_pairs(transfers_pair, block_number):
 
     events = list()
     for pair in pending_transfers_pairs:
-        if block_number > pair.payer_transfer.expiration:
-            assert pair.payee_state == 'payee_expired'
```

Comment:
@hackaugusto can you, please, confirm if it's correct to remove this assert, as we now can expect a `payer_expire` with a paid payee's transfer?

Reply:
Let's think this through. This graph illustrates of all possible states for the transfer pair:

![pair](https://user-images.githubusercontent.com/310139/30872156-8368c7ce-a2bf-11e7-8692-a368ed0c3303.jpg)

The function `set_expired_pairs` must only transition:
- from `pending | pending` to `expired | pending`
- from `expired | pending` or `expired | secret` to `expired | expired`.

This assert is forbidding the states `balance proof | waiting close` and  `balance proof | waiting withdraw`, which is wrong. IOW, it's correct to remove this assertion.

Also:
- The `assert pair.payee_transfer.expiration < pair.payer_transfer.expiration` is done by the sanity check, so it doesn't need to be here.
- The payer should transition to `expired` only if the `payee` is at expired, i.e. `payer_expired` needs fixing.
- We must only use assert for programming errors, IOW assert on invalid pairs that are not part of the above graph. This is better done at the `sanity_check` function, since there are multiple places in the code that change the pair state.